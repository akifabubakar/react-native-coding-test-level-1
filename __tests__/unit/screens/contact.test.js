import { render, fireEvent, screen } from "@testing-library/react-native";
import renderer from "react-test-renderer";

import ContactUsScreen from "../../../src/screens/contact";

describe("<ContactUsScreen/>", () => {
  it("should render correctly", () => {
    const tree = renderer.create(<ContactUsScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should change state if name & email is entered", async () => {
    const component = <ContactUsScreen />;

    render(component);
    const inputName = await screen.findByPlaceholderText("Name");
    const inputEmail = await screen.getByPlaceholderText("Email");

    fireEvent.changeText(inputName, "anon");
    fireEvent.changeText(inputEmail, "anon@gmail.com");

    const newName = await screen.getByTestId("nameTest");
    expect(newName.props.value).toEqual("anon");

    const newEmail = await screen.getByTestId("emailTest");
    expect(newEmail.props.value).toEqual("anon@gmail.com");
  });
});
