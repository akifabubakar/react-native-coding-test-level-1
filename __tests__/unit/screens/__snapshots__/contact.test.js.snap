// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactUsScreen/> should render correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "aliceblue",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "flex": 1,
      }
    }
    keyboardShouldPersistTaps="handled"
    scrollEnabled={false}
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "transparent",
              "flex": 1,
              "justifyContent": "center",
              "width": "60%",
            },
            Object {
              "paddingBottom": 0,
            },
          ]
        }
      >
        <Text>
          Please fill in the required fields
        </Text>
        <TextInput
          onBlur={[Function]}
          onChangeText={[Function]}
          placeholder="Name"
          style={
            Object {
              "backgroundColor": "white",
              "borderRadius": 5,
              "borderWidth": 0.5,
              "fontSize": 16,
              "marginTop": 20,
              "paddingHorizontal": 16,
              "paddingVertical": 8,
              "width": "100%",
            }
          }
          testID="nameTest"
          value=""
        />
        <TextInput
          keyboardType="email-address"
          onBlur={[Function]}
          onChangeText={[Function]}
          placeholder="Email"
          style={
            Object {
              "backgroundColor": "white",
              "borderRadius": 5,
              "borderWidth": 0.5,
              "fontSize": 16,
              "marginTop": 20,
              "paddingHorizontal": 16,
              "paddingVertical": 8,
              "width": "100%",
            }
          }
          testID="emailTest"
          value=""
        />
        <View
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
          >
            <TextInput
              editable={false}
              onPressIn={[Function]}
              placeholder="Birth Date"
              style={
                Object {
                  "backgroundColor": "white",
                  "borderRadius": 5,
                  "borderWidth": 0.5,
                  "fontSize": 16,
                  "marginTop": 20,
                  "paddingHorizontal": 16,
                  "paddingVertical": 8,
                  "width": "100%",
                }
              }
              value=""
            />
          </View>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "cornflowerblue",
              "borderRadius": 20,
              "marginTop": 30,
              "opacity": 1,
              "paddingHorizontal": 20,
              "paddingVertical": 18,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontWeight": "bold",
              }
            }
          >
            Submit
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    transparent={true}
    visible={false}
  />
</RCTSafeAreaView>
`;
